source/board/hexiwear.c

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file    hexiwear.c
+ *
+ * DAPLink Interface Firmware
+ * Copyright (c) 2018, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "target_config.h"
+#include "swd_host.h"
+#include "virtual_fs.h"
+
+const char *board_id = "0214";
+
+// Override default behavior
+//
+// Note - 4 byte alignemnt required as workaround for ARMCC compiler bug with weak references
+__attribute__((aligned(4)))
+const vfs_filename_t daplink_url_name =       "PRODINFOHTM";
+__attribute__((aligned(4)))
+const vfs_filename_t daplink_drive_name =     "Hexiwear";
+__attribute__((aligned(4)))
+const char *const daplink_target_url = "http://www.hexiwear.com/";
+
+#define K_SERIES     0
+#define KW_SERIES    5
+
+void prerun_board_config(void)
+{
+    // SIM peripheral   0x40047000
+    // address offset   0x    1024
+    uint32_t SDID = 0x40048024;
+    uint32_t sdid;
+
+    // get a hold of the target
+    target_set_state(RESET_PROGRAM);
```

Comment:
A reset on boot should be avoided, as this could corrupt any filesystem operations the target is doing.

Reply:
This is required to be able to read the target registers.

Reply:
You should not bring back any filesystem corruption bugs with additional resets at startup.